feat: file skeleton necessary for icicle chart visualization

cguan7 opened the pull request:
🏆 Enhancements

Initial PR to layout file skeleton to implement Icicle Chart

cguan7 committed:
feat: file skeleton necessary for icicle chart visualization

cguan7 committed:
feat: name changes and restructure components

update package to account for clearer name and reorganized implementation for planned features

cguan7 committed:
feat: fix lint errors

khtruong commented on packages/superset-ui-plugin-chart-icicle-event/README.md:
````diff
@@ -0,0 +1,39 @@
+  .register();
+```
+
+Then use it via `SuperChart`. See [storybook](https://apache-superset.github.io/superset-ui-plugins/?selectedKind=plugin-chart-word-cloud) for more details.
````
Might want to update the link in a later PR since IcicleEventChart is not available yet.

khtruong commented on packages/superset-ui-plugin-chart-icicle-event/package.json:
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@superset-ui/plugin-chart-icicle",
```
Might want to re-generate this with icicle-event

khtruong commented on packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx:
```diff
@@ -0,0 +1,47 @@
+import React, { Component, createRef } from 'react';
```
Add file headers. Comment applies to the other files. This should make one of the build pre-checks pass.

cguan7 replied:
Added the file headers to all of the files but the checks are failing due to a timeout

I believe this PR is related and will be a fix for the Netlify checks: 
https://github.com/apache-superset/superset-ui-plugins/pull/160

cguan7 committed:
docs: add license file headers, adjust naming and links

Add Apache license file headers to all files, adjust naming of the viz in the package.json and
update the README to inform that Storybook link is unavailable but will be in the near future
